core/src/main/java/nl/nn/adapterframework/filesystem/LocalFileSystem.java

```diff
@@ -174,66 +160,79 @@ public void createFolder(String folder) throws FileSystemException {
 	@Override
 	public void removeFolder(String folder) throws FileSystemException {
 		if (folderExists(folder)) {
-			toFile(folder).delete();
+			try {
+				Files.delete(toFile(folder));
+			} catch (IOException e) {
+				throw new FileSystemException(e);
+			}
 		}else {
 			throw new FileSystemException("Remove directory for [" + folder + "] has failed. Directory does not exist.");
 		}
 	}
 
 	@Override
-	public File renameFile(File source, File destination) throws FileSystemException {
-		if (!source.renameTo(destination)) {
-			throw new FileSystemException("Could not rename ["+source.getPath()+"] to ["+destination.getPath()+"]");
+	public Path renameFile(Path source, Path destination) throws FileSystemException {
+		try {
+			return Files.move(source, destination);
+		} catch (IOException e) {
+			throw new FileSystemException(e);
 		}
-		return destination;
 	}
 	
 	@Override
-	public File moveFile(File f, String destinationFolder, boolean createFolder) throws FileSystemException {
-		File target = toFile(destinationFolder, f.getName());
-		if (!f.renameTo(target)) {
-			throw new FileSystemException("cannot move file ["+f.getPath()+"] to ["+target.getPath()+"]");
+	public Path moveFile(Path f, String destinationFolder, boolean createFolder) throws FileSystemException {
+		try {
+			return Files.move(f, toFile(destinationFolder, f.getFileName().toString()));
+		} catch (IOException e) {
+			throw new FileSystemException(e);
 		}
-		return target;
 	}
 	@Override
-	public File copyFile(File f, String destinationFolder, boolean createFolder) throws FileSystemException {
-		File target = toFile(destinationFolder, f.getName());
+	public Path copyFile(Path f, String destinationFolder, boolean createFolder) throws FileSystemException {
+		Path target = toFile(destinationFolder, f.getFileName().toString());
 		try {
-			FileUtils.copyFile(f, target);
+			Files.copy(f, target);
 		} catch (IOException e) {
-			throw new FileSystemException("cannot copy file ["+f.getPath()+"] to ["+target.getPath()+"]",e);
+			throw new FileSystemException(e);
```

Comment:
Please add ``"Cannot copy file  ["+ f.getFileName()+"] to ["+ destinationFolder+"]"``